controllers/provisioning_controller.go

```diff
@@ -234,40 +234,13 @@ func (r *ProvisioningReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 
 	if maoOwned {
 		r.Log.V(1).Info("Adding annotation for CBO to take ownership of metal3 deployment created by MAO")
-	}
-
-	// Proceed with creating or updating the Metal3 deployment
-	updated, err := provisioning.EnsureMetal3Deployment(info, metal3DeploymentSelector)
-	if err != nil {
-		return ctrl.Result{}, err
-	}
-	if updated {
-		err = r.Client.Status().Update(context.Background(), baremetalConfig)
-		return ctrl.Result{Requeue: true}, err
-	}
-
-	// Determine the status of the deployment
-	deploymentState, err := provisioning.GetDeploymentState(r.KubeClient.AppsV1(), ComponentNamespace, baremetalConfig)
-	if err != nil {
-		err = r.updateCOStatus(ReasonNotFound, "metal3 deployment inaccessible", "")
-		if err != nil {
-			return ctrl.Result{}, fmt.Errorf("unable to put %q ClusterOperator in Degraded state: %w", clusterOperatorName, err)
-		}
-		return ctrl.Result{}, errors.Wrap(err, "failed to determine state of metal3 deployment")
-	}
-	if deploymentState == appsv1.DeploymentReplicaFailure {
-		err = r.updateCOStatus(ReasonDeployTimedOut, "metal3 deployment rollout taking too long", "")
-		if err != nil {
-			return ctrl.Result{}, fmt.Errorf("unable to put %q ClusterOperator in Degraded state: %w", clusterOperatorName, err)
-		}
-	} else if deploymentState == appsv1.DeploymentAvailable {
-		err = r.updateCOStatus(ReasonSyncing, "metal3 pod running", "starting other metal3 services")
-		if err != nil {
-			return ctrl.Result{}, fmt.Errorf("unable to put %q ClusterOperator in Progressing state: %w", clusterOperatorName, err)
-		}
+		info.LabelSelector = metal3DeploymentSelector
```

Comment:
Done. Reason was for not updating it unconditionally was because we are expected to have a different value from the CBO generated value only when we are trying to preserve the MAO deployment.